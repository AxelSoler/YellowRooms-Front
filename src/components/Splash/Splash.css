.splash-page {
  background-color: #ffde59;
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
  text-decoration: none;
}

.welcome {
  color: #fff;
  font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
  font-size: 40px;
  text-shadow: 4px 3px 5px #737372;
}

.mainLogo {
  width: 270px;
}

.loginInput {
  width: 80vw;
  height: 36px;
  color: #fff;
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
  font-size: 24px;
  background-color: #ffde59;
  margin-bottom: 3%;
  padding: 12px;
  border: 4px #737372;
  border-style: dashed;
  border-radius: 16px;
}

.loginInput:focus {
  background: #d0b447;
}

.register-link {
  text-decoration: none;
  width: 30vw;
  padding: 12px;
  color: #fff;
  background-color: #91bd04;
  border-radius: 20px;
  font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
  font-size: 18px;
}

.register-link:active {
  background-color: #6f9004;
}

.login-form {
  margin-bottom: 24px;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.loginBtn {
  width: 80vw;
  padding: 12px;
  color: #fff;
  background-color: #91bd04;
  border: 4px #737372;
  border-radius: 20px;
  font-family: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
  font-size: 22px;
  cursor: pointer;
}

.loginBtn:active {
  background-color: #6f9004;
}

.title-splash {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 90%;
  margin: 0 auto 3vh auto;
}

@media only screen and (min-width: 769px) {
  .login-form {
    width: 80%;
  }

  .register-link {
    min-width: 130px;
    width: 10%;
    font-size: 24px;
  }

  .loginBtn {
    width: 20vw;
    font-size: 30px;
  }

  .loginInput {
    width: 45vw;
  }

  .title-splash {
    width: 48vw;
  }

  .welcome {
    text-align: left;
  }

  .mainLogo {
    width: 350px;
  }
}
