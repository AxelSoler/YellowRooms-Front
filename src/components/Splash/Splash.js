import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import axios from 'axios';
import mainLogo from '../../img/YELLOW-3.png';
import './Splash.css';

const Splash = () => {
  const [userName, setUserName] = useState('');
  const [userPassword, setUserPassword] = useState('');
  const loginURL = 'http://localhost:3001/users';

  const userSubmit = async (e) => {
    e.preventDefault();
    axios.get(loginURL).then((response) => {
      const data = response.data
        .filter((item) => item.name === userName && item.password === userPassword);
      if (data[0]) {
        setUserName('');
        setUserPassword('');
        window.location.href = '/Rooms';
      }
    });
  };
  return (
    <div className="splash-page">
      <img src={mainLogo} alt="mainLogo" className="mainLogo" />
      <form className="login-form" onSubmit={userSubmit}>
        <div className="title-splash">
          <h2 className="welcome">WELCOME</h2>
          <NavLink className="register-link" to="signup">REGISTER</NavLink>
        </div>
        <input onChange={(e) => setUserName(e.target.value)} className="loginInput" type="text" placeholder="User" required />
        <input onChange={(e) => setUserPassword(e.target.value)} className="loginInput" type="text" placeholder="Password" required />
        <button className="loginBtn" type="submit">LOGIN</button>
      </form>
    </div>
  );
};

export default Splash;
