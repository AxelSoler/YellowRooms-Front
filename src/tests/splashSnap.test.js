import React from "react";
import renderer from "react-test-renderer";
import Splash from "../components/Splash/Splash";
import wrapper from "./wrapMeInValidComponents";
test("splash snapshot", () => {
  const tree = renderer.create(wrapper(<Splash />)).toJSON();

  expect(tree).toMatchSnapshot();
});
