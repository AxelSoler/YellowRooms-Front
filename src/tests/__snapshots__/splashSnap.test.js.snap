// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splash snapshot 1`] = `
Array [
  <section
    className="splash-page"
  >
    <img
      alt="mainLogo"
      className="mainLogo"
      src="YELLOW-3.png"
    />
    <section>
      <button
        className="registerBtn"
        id="loginBtn"
        onClick={[Function]}
        type="button"
      >
        LOGIN
      </button>
      <div
        className="modalInactive"
      >
        <form
          className="login-form"
          onSubmit={[Function]}
        >
          <div
            className="signupTitle"
          >
            <h3
              className="signup"
            >
              LogIn
            </h3>
            <button
              className="closeBtn"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="crossBtn"
                fill="currentColor"
                height="24px"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                version="1.1"
                viewBox="0 0 16 16"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.854 12.854c-0-0-0-0-0-0l-4.854-4.854 4.854-4.854c0-0 0-0 0-0 0.052-0.052 0.090-0.113 0.114-0.178 0.066-0.178 0.028-0.386-0.114-0.529l-2.293-2.293c-0.143-0.143-0.351-0.181-0.529-0.114-0.065 0.024-0.126 0.062-0.178 0.114 0 0-0 0-0 0l-4.854 4.854-4.854-4.854c-0-0-0-0-0-0-0.052-0.052-0.113-0.090-0.178-0.114-0.178-0.066-0.386-0.029-0.529 0.114l-2.293 2.293c-0.143 0.143-0.181 0.351-0.114 0.529 0.024 0.065 0.062 0.126 0.114 0.178 0 0 0 0 0 0l4.854 4.854-4.854 4.854c-0 0-0 0-0 0-0.052 0.052-0.090 0.113-0.114 0.178-0.066 0.178-0.029 0.386 0.114 0.529l2.293 2.293c0.143 0.143 0.351 0.181 0.529 0.114 0.065-0.024 0.126-0.062 0.178-0.114 0-0 0-0 0-0l4.854-4.854 4.854 4.854c0 0 0 0 0 0 0.052 0.052 0.113 0.090 0.178 0.114 0.178 0.066 0.386 0.029 0.529-0.114l2.293-2.293c0.143-0.143 0.181-0.351 0.114-0.529-0.024-0.065-0.062-0.126-0.114-0.178z"
                />
              </svg>
            </button>
          </div>
          <input
            className="loginInput"
            name="login"
            onChange={[Function]}
            placeholder="User"
            required={true}
            type="text"
            value=""
          />
          <input
            className="loginInput"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
          <button
            className="registerBtn"
            type="submit"
          >
            LogIn
          </button>
        </form>
      </div>
    </section>
    <section>
      <button
        className="registerBtn"
        id="signBtn"
        onClick={[Function]}
        type="button"
      >
        REGISTER
      </button>
      <div
        className="modalInactive"
      >
        <div
          className="signupTitle"
        >
          <h3
            className="signup"
          >
            SignUp
          </h3>
          <button
            className="closeBtn"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="crossBtn"
              fill="currentColor"
              height="24px"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              version="1.1"
              viewBox="0 0 16 16"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.854 12.854c-0-0-0-0-0-0l-4.854-4.854 4.854-4.854c0-0 0-0 0-0 0.052-0.052 0.090-0.113 0.114-0.178 0.066-0.178 0.028-0.386-0.114-0.529l-2.293-2.293c-0.143-0.143-0.351-0.181-0.529-0.114-0.065 0.024-0.126 0.062-0.178 0.114 0 0-0 0-0 0l-4.854 4.854-4.854-4.854c-0-0-0-0-0-0-0.052-0.052-0.113-0.090-0.178-0.114-0.178-0.066-0.386-0.029-0.529 0.114l-2.293 2.293c-0.143 0.143-0.181 0.351-0.114 0.529 0.024 0.065 0.062 0.126 0.114 0.178 0 0 0 0 0 0l4.854 4.854-4.854 4.854c-0 0-0 0-0 0-0.052 0.052-0.090 0.113-0.114 0.178-0.066 0.178-0.029 0.386 0.114 0.529l2.293 2.293c0.143 0.143 0.351 0.181 0.529 0.114 0.065-0.024 0.126-0.062 0.178-0.114 0-0 0-0 0-0l4.854-4.854 4.854 4.854c0 0 0 0 0 0 0.052 0.052 0.113 0.090 0.178 0.114 0.178 0.066 0.386 0.029 0.529-0.114l2.293-2.293c0.143-0.143 0.181-0.351 0.114-0.529-0.024-0.065-0.062-0.126-0.114-0.178z"
              />
            </svg>
          </button>
        </div>
        <form
          onSubmit={[Function]}
        >
          <input
            className="loginInput"
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="User"
            required={true}
            type="text"
          />
          <input
            className="loginInput"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="email"
          />
          <input
            className="loginInput"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
          />
          <button
            className="registerBtn"
            type="submit"
          >
            Register
          </button>
        </form>
      </div>
    </section>
    <footer
      className="footer"
    >
      <li
        className="profile"
      >
        Microverse Final Capstone Team
      </li>
      <ul
        className="ulProfile"
      >
        <li>
          <a
            className="profile"
            href="https://github.com/marurevi"
            rel="noopener noreferrer"
            target="_blank"
          >
            Mariana
          </a>
        </li>
        <li>
          <a
            className="profile"
            href="https://github.com/anagudelogu"
            rel="noopener noreferrer"
            target="_blank"
          >
            Andres
          </a>
        </li>
        <li>
          <a
            className="profile"
            href="https://github.com/AxelSoler"
            rel="noopener noreferrer"
            target="_blank"
          >
            Axel
          </a>
        </li>
        <li>
          <a
            className="profile"
            href="https://github.com/Tadesse-Alemayehu"
            rel="noopener noreferrer"
            target="_blank"
          >
            Tadesse
          </a>
        </li>
      </ul>
    </footer>
  </section>,
  ";",
]
`;
